### PR TITLE
Metal building pipeline enhancement

### PR DESCRIPTION
**Pull request by yaroslavyaroslav**
1. Metal shaders for macOS compile with `build.rs`
2. `metallib` stored within std out dir rather than in the git repo under hardcoded path.
3. Prebuilt library and script deleted.
4. `build.rs` handles conditional debug symbols injection into metal lib by setting `[profile.release]\n debug = true` in `Cargo.toml`

### CODE REVIEW

**Comment by yaroslavyaroslav** [comment]
PR is converted to draft due to iOS pipeline haven't been implemented yet.

**Comment by yaroslavyaroslav** [comment]
@moven0831 I haven't removed all.metal file yet, to avoid unnecessary conflicts with your pending PR #13 But it's the matter of changing a single function in the build current pipeline, so it won't take any long to do it afterwards.

PR is ready for review

File: mopro-msm/build.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+fn compile_shaders() {
+    let shader_dir = "src/msm/metal/shader/";
+    let out_dir = env::var("OUT_DIR").unwrap();
+
+    // List your Metal shaders here.
+    let shaders = vec!["all.metal"];
+
+    let shaders_to_check = vec!["all.metal", "msm.h.metal"];
```
Comment (moven0831) [inline mopro-msm/build.rs:14]: will replace the `all.metal` with refactored shaders to be compiled

**Review by moven0831 — APPROVED** [review]
LGTM! and also this is compatible with PR #13 since the main purpose of it is to provide a new metal ec backend instead of the overall metal msm

**Pull request merged**